ui/appui-react/src/appui-react/statusfields/unitsfield/UnitsField.tsx

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import "./UnitsField.scss";
+import * as React from "react";
+import type { CommonProps } from "@itwin/core-react";
+import { SvgLightbulbHollow } from "@itwin/itwinui-icons-react";
+import type { SelectOption } from "@itwin/itwinui-react";
+import { IconButton, List, ListItem } from "@itwin/itwinui-react";
+import { TitleBar } from "../../layout/footer/dialog/TitleBar";
+import { StatusBar } from "../../statusbar/StatusBar";
+import { FooterIndicator } from "../../layout/footer/Indicator";
+import { IModelApp } from "@itwin/core-frontend";
+import type { UnitSystemKey } from "@itwin/core-quantity";
+
+/** Props for [[UnitsField]].
+ * @internal
+ */
+interface UnitsFieldProps extends CommonProps {
+  label?: string;
+  title: string;
+  options: SelectOption<string>[];
+}
+
+/** Component used for setting the active unit system.
+ * @internal
+ */
+export function UnitsField(props: UnitsFieldProps) {
```

Comment:
As discussed, let's keep this in the test-app (appui-test-providers).

Reply:
Done